### PR TITLE
Data-Layer: local actions re-issue bug

### PR DESCRIPTION
**Pull request by rodrigoi**
This PR adds checks to `wpcom-api-middleware` to prevent actions to be passed along for actions with a data layer progress property.

It also makes minor changes by fixing the unit tests. The assumption for those fixes is that the tests are conceptually correct.

to test:

```sh
npm run test-client client/state/data-layer/test/wpcom-api-middleware.js
```

### CODE REVIEW

File: client/state/data-layer/wpcom-api-middleware.js [hunk 1]
```diff
@@ -17,16 +17,16 @@ const mergedHandlers = mergeHandlers(
...
 	const meta = action.meta;
 	if ( ! meta ) {
-		return true;
+		return false;
 	}
 
 	const data = meta.dataLayer;
 	if ( ! data ) {
-		return true;
+		return false;
```
Comment (dmsnell) [inline client/state/data-layer/wpcom-api-middleware.js]: Why is this changed to `false` here and above? If they have no meta don't we definitely want to pass them along the middleware chain? Only _if_ it has meta _and_ the identified network meta do we not pass it along…

Comment (rodrigoi) [inline client/state/data-layer/wpcom-api-middleware.js]: I've made those changes to make the tests pass once the `beenCalled` issue was detected, assuming that the tests were conceptually correct. We can always change the tests.

Comment (dmsnell) [inline client/state/data-layer/wpcom-api-middleware.js]: I think we should leave these here because we should definitely pass an action through to the next middleware if there is no `meta` on it. we need to evaluate the tests if they are wrong.

**Comment by rodrigoi** [comment]
@dmsnell created this new PR to work on the `data-layer` tests: https://github.com/Automattic/wp-calypso/pull/15793

**Comment by gwwar** [comment]
@rodrigoi do you mind keeping 15730 and 15793 together, or try merging 15793 into this branch instead of master? I think we wouldn't be able to land a green branch otherwise.

File: client/state/data-layer/wpcom-api-middleware.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ const shouldNext = action => {
 	}
 
 	// is a network response, don't reissue
-	if ( data.data || data.error || data.headers ) {
+	if ( data.data || data.error || data.headers || data.progress ) {
```
Comment (gwwar) [inline client/state/data-layer/wpcom-api-middleware.js]: We can simplify this to the following, but I'm also fine with the current approach.
```jsx
return ! ( data.data || data.error || data.headers || data.progress );
```

**Comment by gwwar** [comment]
The changes here look good @rodrigoi 👍  

Since we're updating the data-layer, @alisterscott or @hoverduck would it be possible to run e2e tests against this branch?

**Comment by gwwar** [comment]
AT still looks like it works properly on this branch 👍

**Comment by hoverduck** [comment]
> Since we're updating the data-layer, @alisterscott or @hoverduck would it be possible to run e2e tests against this branch?

e2e tests ran clean and green

**Comment by gwwar** [comment]
Thanks a ton @hoverduck !

File: client/state/data-layer/wpcom-api-middleware.js (hunk 2 above)
Comment (dmsnell) [inline client/state/data-layer/wpcom-api-middleware.js]: the reason I originally didn't simplify was to keep the pattern of the function present. as in, we have a list of conditions which can accept, reject, or do nothing. this is just the last one.

simplifying is okay, but I feel like the only benefit is a few fewer bytes transferred because it obscures the flow (at least in my head). heck, this probably already gets optimized to the simpler version

Comment (gwwar) [inline client/state/data-layer/wpcom-api-middleware.js]: I'm fine with the original here 👍

**Comment by rodrigoi** [comment]
@gwwar @hoverduck and @dmsnell thanks for the help!
